server/resp/src/main/java/org/infinispan/server/resp/serialization/Resp3Type.java

```diff
@@ -69,6 +72,23 @@ public void serialize(Object object, ResponseWriter writer) {
       public void serialize(Object object, ResponseWriter writer) {
          writer.doubles((Number) object);
       }
+   },
+
+   AUTO {
+      @Override
+      public void serialize(Object object, ResponseWriter writer) {
+         if (object instanceof CharSequence s) {
+            writer.string(s);
+         } else if (object instanceof Number n) {
+            writer.integers(n);
+         } else if (object instanceof Collection<?> c) {
+            writer.array(c, AUTO);
+         } else if (object instanceof Map<?,?> m) {
+            writer.map(m, AUTO);
+         } else {
+            throw new UnsupportedOperationException();
```

Comment:
It should be possible to invoke `writer.serialize(object)` if it is unknown. But that could cover an underlying issue.